### ISSUE TITLE
Queue: When a column has a filter applied, AT does not read to users that a filter is applied. 

### ISSUE THREAD

**Issue by alisan16** [department-of-veterans-affairs/caseflow#12134]
The state of the Filter elements Task, Types and Docket is not read by assistive technology programs. If a filter is applied, visually a user can see this change has been applied to the data, however this change is not read by assistive technology programs. It should read "Column_name filtered by filter_criteria". (New: 08/21/2019) in caseflowdemo (under Queue/your queue/assigned)

**Comment by lowellrex** [comment]
![image](https://user-images.githubusercontent.com/32683958/67010956-adafac80-f0bc-11e9-8c15-4f5f29cabb8a.png)

Is the issue here that the "Filtering by: Case Type (1)" text is not being read, or that when the "Type(s)" column heading is read, there is no indication that a filter has been applied?

**Comment by alisan16** [comment]
I believe it is the latter, that when the column heading is read, the state of the filter isn't being described. There was also an open question of whether the filter itself could be applied using keyboard navigation (primarily tab, and ctrl+alt+arrow within a table). cc @carodew 

This was the description provided to us: 

"The state of the Filter elements Task, Types and Docket is not read by assistive technology programs. If a filter is applied, visually a user can see this change has been applied to the data, however this change is not read by assistive technology programs. (New: 08/21/2019) in caseflowdemo (under Queue/your queue/assigned) – I couldn’t get it to function in Chrome"
<img width="999" alt="Screen Shot 2019-10-17 at 10 48 38 AM" src="https://user-images.githubusercontent.com/1198106/67020157-bf4c8080-f0cb-11e9-9aea-ad32efa814bd.png">

**Comment by hschallhorn** [comment]
@alisan16 What should we return for multiple filter selections? Should we add "and"s in between? Build a grammatically correct list? What's your take on the oxford comma?

**Comment by alisan16** [comment]
Completely up to you! Either of those options sounds good to me.

**Comment by araposo-tistatech** [comment]
reopening as this has not been resolved per the latest audit: 

"I couldn’t get it to function in Chrome  (3/26/2020 I was able to get to the filter but it was not read by JAWs once I activated (checked the box) of a filter)"

**Comment by ajspotts** [comment]
@araposo-tistatech Using Chrome, JAWS was able to successfully read the filter option.  In the screenshot you can see the red box (which is indicative of JAWS focus) highlighting the filter.   

<img width="1258" alt="Screen Shot 2020-06-16 at 10 43 02 AM" src="https://user-images.githubusercontent.com/18618189/84791908-42a6d480-afc1-11ea-915e-293935a2a467.png">

**Comment by araposo-tistatech** [comment]
@chandracarney please see Alec's note above and advise how we should move forward. We are unable to replicate the issue.

**Comment by carodew** [comment]
I think what this is saying is that when a filter is currently in effect that the screen reader should announce what filter is applied when the header cell for that column receives focus, so that the user knows that there is a filter applied (versus the table showing all data). Worth confirming with Linsey, though.

**Comment by ajspotts** [comment]
I believe there are 2 separate issues, 1 from the original audit (listed in Alisa's issue description) and then the issue that Ann-Marie mentioned from the most recent audit when she re-opened this issue.  Alisa's issue describing the problem with the column header was taken care of in a [PR](https://github.com/department-of-veterans-affairs/caseflow/pull/13049) from Hunter.  Ann-Marie's issue from the newer audit (filter not read by JAWS when activated) is what I was referencing.  Hope that provides some clarity instead of more confusion!  I'm adding the full description of the original and updated issues from the Caseflow a11y issues [doc](https://docs.google.com/document/d/1iMazQsgr9hXZ7pPP4Q2RH5qKN-Xz8IUHGHo0bPSubqw/edit?ts=5ea9cd2e#):

> The state of the Filter elements Task, Types and Docket is not read by assistive technology programs. If a filter is applied, visually a user can see this change has been applied to the data, however this change is not read by assistive technology programs. (New: 08/21/2019) in caseflowdemo (under Queue/your queue/assigned) – I couldn’t get it to function in Chrome  (3/26/2020 I was able to get to the filter but it was not read by JAWs once I activated (checked the box) of a filter)

**Comment by chandracarney** [comment]
@ajspotts Thanks for checking in! I haven't had a chance to test this out on my own, but I do wonder if this part was taken care of by @sahalliburton's recent changes in #14513? Though not yet merged, I know there was a big overhaul of how tables work for screen readers. 

If not, I am happy to 1) make some time to test myself to understand behavior, and 2) check in with Linsey for more context or and test if this has been resolved.

**Comment by alisan16** [comment]
Potential duplicate of #12902, blocked until we get further confirmation from Section 508

**Comment by ajspotts** [comment]
> Potential duplicate of #12902, blocked until we get further confirmation from Section 508

@alisan16 @jimruggiero I see that we've closed the other issue that was also awaiting confirmation.  Do we know if we can go ahead and close this one as well?

**Comment by jimruggiero** [comment]
Hi Alec,

Thanks for the follow up. We don't know yet. I will reach out to Linsey
and let you know when I hear back.

Thanks,
Jim

**Comment by araposo-tistatech** [comment]
@ajspotts Lindsey has confirmed that this is not a duplicate. In this case the state needs to be read that a filter is applied for 12902 the issue is about being accessible via the keyboard.

**Comment by ajspotts** [comment]
Closing after confirming with @araposo-tistatech during JAWS testing walkthrough that the filter options are correctly being announced
